python/lsst/ts/ess/common/abstract_command_handler.py

```diff
@@ -196,10 +206,97 @@ async def stop_sending_telemetry(self) -> None:
                 msg="Not started yet. Ignoring stop command.",
                 response_code=ResponseCode.NOT_STARTED,
             )
-        await self.disconnect_devices()
+
+        while self._devices:
+            device: BaseDevice = self._devices.pop(-1)
+            self.log.debug(f"Closing {device} device with name {device.name}")
+            await device.close()
+
         self._started = False
 
-    @abstractmethod
-    async def disconnect_devices(self) -> None:
-        """Loop over the configuration and start all devices."""
-        raise NotImplementedError
+    def get_device(
+        self, device_configuration: typing.Dict[str, typing.Any]
+    ) -> BaseDevice:
+        """Get the device to connect to by using the specified configuration.
+
+        Parameters
+        ----------
+        device_configuration: `dict`
+            A dict representing the device to connect to. The format of the
+            dict follows the configuration of the ts_ess_csc project.
+
+        Returns
+        -------
+        device: `common.device.BaseDevice`
+            The device to connect to.
+
+        Raises
+        ------
+        RuntimeError
+            In case an incorrect configuration has been loaded.
+
+        Notes
+        -----
+        In this case a MockDevice always is returned. Sub-classes should
+        override this method to add support for other devices.
+        """
+        sensor = self.get_sensor(device_configuration=device_configuration)
+        self.log.debug(
+            f"Creating MockDevice with name {device_configuration[Key.NAME]} and sensor {sensor}"
+        )
+        device: BaseDevice = MockDevice(
+            name=device_configuration[Key.NAME],
+            device_id=device_configuration[Key.FTDI_ID],
+            sensor=sensor,
+            callback_func=self._callback,
+            log=self.log,
+            disconnected_channel=None,
+        )
+        return device
+
+    def get_sensor(
```

Comment:
Please consider:
* Renaming this to `make_sensor` (it creates a sensor, it doesn't get one that has been created) 
* Making it a free function (the only thing it uses from the class is the log). I suggest making a sensor registry below, and this could easily become a free function in the module that defines the registry.

Reply:
I decided to do this in this ticket anyway.